src/common/transformations/src/transformations/common_optimizations/lstm_cell_fusion.cpp

```diff
@@ -352,3 +446,116 @@ ov::pass::LSTMCellFusion::LSTMCellFusion() {
+        const auto& h = pattern_map.at(h_label);
+        const auto& c = pattern_map.at(c_label);
+
+        const auto& wi = pattern_map.at(wi_label);
+        const auto& ri = pattern_map.at(ri_label);
+        const auto& bi = pattern_map.at(bi_label);
+
+        const auto& wf = pattern_map.at(wf_label);
+        const auto& rf = pattern_map.at(rf_label);
+        const auto& bf = pattern_map.at(bf_label);
+
+        const auto& wo = pattern_map.at(wo_label);
+        const auto& ro = pattern_map.at(ro_label);
+        const auto& bo = pattern_map.at(bo_label);
+
+        const auto& wc = pattern_map.at(wc_label);
+        const auto& rc = pattern_map.at(rc_label);
+        const auto& bc = pattern_map.at(bc_label);
+
+        const auto& it = pattern_map.at(it_label).get_node_shared_ptr();
+        const auto& ft = pattern_map.at(ft_label).get_node_shared_ptr();
+        const auto& ot = pattern_map.at(ot_label).get_node_shared_ptr();
+        const auto& c1t = pattern_map.at(c1t_label).get_node_shared_ptr();
+        const auto& ct_activated = pattern_map.at(ct_activated_label).get_node_shared_ptr();
+
+        auto ct = pattern_map.at(ct_label);
+        auto ht = pattern_map.at(ht_label);
+
+        // check weights format
+        auto input_size = static_cast<size_t>(wi.get_shape()[0]);
+        auto hidden_size = static_cast<size_t>(wi.get_shape()[1]);
+        bool weights_format_ok =
+            check_weights_format(wi, ri, bi, input_size, hidden_size, pattern_map, x_by_wi_label, h_by_ri_label);
+        weights_format_ok &=
+            check_weights_format(wf, rf, bf, input_size, hidden_size, pattern_map, x_by_wf_label, h_by_rf_label);
+        weights_format_ok &=
+            check_weights_format(wo, ro, bo, input_size, hidden_size, pattern_map, x_by_wo_label, h_by_ro_label);
+        weights_format_ok &=
+            check_weights_format(wc, rc, bc, input_size, hidden_size, pattern_map, x_by_wc_label, h_by_rc_label);
+        if (!weights_format_ok) {
+            return false;
+        }
+
+        // check activation functions
+        std::string f_activation_name = it->get_type_name();
+        if (f_activation_name != ft->get_type_name() || f_activation_name != ot->get_type_name()) {
+            return false;
+        }
+        f_activation_name[0] = std::tolower(f_activation_name[0]);
+
+        std::string g_activation_name = get_activation_name(c1t);
+        std::string h_activation_name = get_activation_name(ct_activated);
+
+        // prepare weights in a format required by LSTMCell
+        // W weights must be in fico format
+        ov::Output<ov::Node> W = prepare_weight_fico(wf, wi, wc, wo, rg);
+        ov::Output<ov::Node> R = prepare_weight_fico(rf, ri, rc, ro, rg);
+
+        ov::Output<ov::Node> B = rg.make<ov::op::v0::Concat>(ov::OutputVector{bf, bi, bc, bo}, 0);
+        if (const auto& constant = ov::util::constantfold_subgraph(B))
+            B = constant;
```

Comment:
There is a ConstantFolding transformation in MOC after this transformation. Do we need it here?

Reply:
Yes, let us have it by the same way like in the other transformations.